scripts/nice-history/nice-history.js

```diff
@@ -0,0 +1,220 @@
+        userID = document.getElementById('core').attributes['data-user'].value;
+        console.log('Nice watcher detected user: '+userID);
+      }
+      else {
+        console.error('Nice watcher failed to detect user');
+      }
+    }
+    return userID;
+  }
+
+  const niceWatch = () => {
+    // Detect if a nice has happened
+    let niceEls = document.querySelectorAll('body > .nice_toast');
+    if(niceEls) {
+      niceEls.forEach(niceEl => {
+          // don't detect nices in the display window
+          let nice = niceEl.innerHTML;
+          console.warn('Nice detected!');
+          console.log(nice);
+          storeNice(nice);
+      });
+    }
+  }
+
+  const storeNice = (nice) => {
+    if(!getUserID()) {
+      console.error('Failed to store nice: user ID not detected');
+      return;
+    }
+
+    // Prepare nice for storage
+    let userName = '<UNKNOWN>';
+    try {
+      userName = nice.substr(nice.indexOf('</a>')+4,nice.indexOf('<br>')-nice.indexOf('</a')-4);
+    } catch(err) {
+      console.error('Failed to parse nice sender');
+    }
+    if(recentNices.indexOf(userName)>=0) {
+      console.log('Skipping repeat nice entry: '+userName);
+      return;
+    }
+    // Attempt to build the URL-friendly name for link purposes
+    let normalizedName = '';
+    try {
+      normalizedName = userName.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
+      normalizedName = normalizedName.replace(' ','_');
+      normalizedName = normalizedName.replace('.',':');
+    } catch(err) {
+      console.warn('Failed to normalize username: '+userName);
+    }
+    let newNice = {
+      'user': userName,
+      'profile': normalizedName,
+      'date': new Date().getTime()
+      //,'content': nice  // for testing only, increases cookie size
+    };
+
+    // Append to old nices and save
+    let niceObjects = getStoredNices();
+    niceObjects.push(newNice);
+    recentNices.push(userName);
+    localStorage.setItem(getUserID()+'.NiceWatcher', JSON.stringify(niceObjects));
+  }
+
+  const getStoredNices = () => {
+    let niceObjects = [];
+    let existingNices = localStorage.getItem(getUserID()+'.NiceWatcher');
+    if(existingNices) {
+      try {
+        niceObjects = JSON.parse(existingNices);
+      } catch(err) {
+        console.error('Failed to decode existing nice data: '+existingNices);
+        console.error(err);
+      }
+    }
+    return niceObjects;
+  }
+
+  const addNiceUI = () => {
+    let niceCount = document.getElementById('nicecount').parentElement;
+    let niceButton = document.createElement('button');
+    niceButton.innerText = '🕒';
+    niceButton.style = 'margin-left:0.5em';
+    niceButton.onclick = openNiceUI;
+    niceCount.append(niceButton);
+  }
+
+  const openNiceUI = () => {
+    // create modal and open
+    let niceWindow = document.createElement('div');
+    niceWindow.className = 'dialog';
+    let dialogContents = '<div><div><div><h3>Nice History</h3><div style="margin:0.5em;">'
```

Comment:
I should probably provide some documentation somewhere on how PFQ's code works for particular things. Again, this is relying on a particular HTML structure that may or may not change in future. Instead you might have better results with PFQ's `Dialog` constructor:

```js
const buttons = [];
buttons.push({text:'Label', action:doSomething});
buttons.push("cancel");
const dialog = new Dialog("Title", "Contents", buttons);
dialog.opened(doSomethingWithCreatedDialog);
dialog.oncancel(doSomethingIfDialogIsCancelled);
```
This ensures that things like the overlay and scroll lock are handled just like PFQ's existing dialog boxes (which, in future, will be actual `<dialog>` elements!)